mlrun/utils/notifications/notification_pusher.py

```diff
@@ -392,17 +392,27 @@ def get_workflow_steps(self, run: mlrun.model.RunObject) -> list:
         steps = []
         db = mlrun.get_run_db()
 
-        def _add_run_step(_node_name, _node_template, _step_kind):
-            _run = db.list_runs(
-                project=run.metadata.project,
-                labels=f"mlrun/runner-pod={_node_name}",
-            )[0]
-            _run["step_kind"] = _step_kind
+        def _add_run_step(_step):
+            try:
+                _run = db.list_runs(
+                    project=run.metadata.project,
+                    labels=f"mlrun/runner-pod={_step.node_name}",
+                )[0]
+            except IndexError:
+                _run = {
+                    "metadata": {
+                        "name": _step.display_name,
+                        "project": run.metadata.project,
+                    },
+                }
+            _run["step_kind"] = _step.step_type
+            if _step.skipped:
+                _run.setdefault("status", {})["state"] = "skipped"
```

Comment:
Use `mlrun.common.runtimes.constants.RunStates`. Looks like you need to add skipped there.
Also we might want to check with UI how they show skipped state. I'm not sure if it's something we already have or new here.